pangeo_forge_recipes/recipes/xarray_zarr.py

```diff
@@ -592,6 +592,20 @@ def store_chunk(
             var_coded.attrs = {}
             with dask.config.set(scheduler="single-threaded"):  # make sure we don't use a scheduler
                 var = xr.backends.zarr.encode_zarr_variable(var_coded)
+                logger.debug(
+                    f"Converting variable {vname} of {var.data.nbytes} bytes to `numpy.ndarray`"
+                )
+                if var.data.nbytes > 500_000_000:
```

Comment:
Rather than hard-coding this, I see two options for more flexibility:

- Try to examine the host environment dynamically and figure out how much memory is available. (Dask does such things often).
- Use an environment variable such as `PANGEO_FORGE_MAX_MEMORY`

Reply:
> Try to examine the host environment dynamically and figure out how much memory is available. (Dask does such things often).

This was my first thought but could figure out a simple implementation. Are you aware of any (relatively) simple ways to do this that don't add external dependencies? I was looking at `psutil` (maybe unmaintained now?) and the like, but thought it wasn't worth adding a dependency just for this?

> Use an environment variable such as PANGEO_FORGE_MAX_MEMORY

For this option, maybe something like:

```python
import os
from typing

_MAX_MEMORY = (
    int(os.environ["PANGEO_FORGE_MAX_MEMORY"])  # i think env variables come in as strings?
    if os.environ["PANGEO_FORGE_MAX_MEMORY"]
    else 500_000_000
)
```
?

Reply:
Went with the env variable option in https://github.com/rabernat/pangeo-forge/pull/2/commits/7d9665cdfe424e6ee5bc4b34ef52e68524d932c1 because it seemed vastly easier to implement.